jenkins/macros.j2

```diff
@@ -18,3 +18,52 @@
 {% macro per_exec_ws(folder) -%}
   "workspace/exec_${env.EXECUTOR_NUMBER}/{{ folder }}"
 {%- endmacro -%}
+
+{% macro sharded_test_step(name, num_shards, node, ws) %}
+{% for shard_index in range(1, num_shards + 1) %}
+  '{{ name }} {{ shard_index }} of {{ num_shards }}': {
```

Comment:
minor nit: want to 0-pad shard_index and num_shards here?

Reply:
meh, this is intended for humans and everything is `text-align: center`-ed anyways so 0-padding won't make it easier to read IMO (also we are at like 3 shards max not 10 so we can revisit later)

Reply:
ok sg. the main intent was that Jenkins sorts these alphabetically iiuc.

Reply:
ah I see, we should still be ok on that front but would probably need to pad if we shard more than n=9